bundles/org.openhab.core.config.discovery.usbserial.otheros/src/main/java/org/openhab/core/config/discovery/usbserial/otheros/internal/OtherOSUsbSerialDiscovery.java

```diff
@@ -0,0 +1,221 @@
+        Set<UsbSerialDeviceInformation> removed = setDifference(lastScanResult, scanResult);
+        Set<UsbSerialDeviceInformation> unchanged = setDifference(scanResult, added);
+
+        lastScanResult = scanResult;
+
+        removed.stream().forEach(this::announceRemovedDevice);
+        added.stream().forEach(this::announceAddedDevice);
+        unchanged.stream().forEach(this::announceAddedDevice);
+    }
+
+    private <T> Set<T> setDifference(Set<T> set1, Set<T> set2) {
+        Set<T> result = new HashSet<>(set1);
+        result.removeAll(set2);
+        return result;
+    }
+
+    @Override
+    public void registerDiscoveryListener(UsbSerialDiscoveryListener listener) {
+        discoveryListeners.add(listener);
+        for (UsbSerialDeviceInformation deviceInfo : lastScanResult) {
+            listener.usbSerialDeviceDiscovered(deviceInfo);
+        }
+    }
+
+    @Override
+    public void unregisterDiscoveryListener(UsbSerialDiscoveryListener listener) {
+        discoveryListeners.remove(listener);
+    }
+
+    /**
+     * Traverse the USB tree for devices that are children of the ROOT hub, and return a set of USB device information.
+     * 
+     * @param usbHub the hub whose children are to be found.
+     * @return a set of USB device information.
+     */
+    private Set<UsbSerialDeviceInformation> scanAllUsbDevicesInformation() {
+        try {
+            return scanChildUsbDeviceInformation(UsbHostManager.getUsbServices().getRootUsbHub());
+        } catch (SecurityException | UsbException e) {
+            logger.warn("Error getting USB device information: {}", e.getMessage());
+            return Set.of();
+        }
+    }
+
+    /**
+     * Traverse the USB tree for devices that are children of the given hub, and return a set of USB device information.
+     * 
+     * @param usbHub the hub whose children are to be found.
+     * @return a set of USB device information.
+     */
+    private Set<UsbSerialDeviceInformation> scanChildUsbDeviceInformation(UsbHub usbHub) {
+        Set<UsbSerialDeviceInformation> result = new HashSet<>();
+
+        @SuppressWarnings("unchecked")
+        List<UsbDevice> deviceList = usbHub.getAttachedUsbDevices();
+
+        deviceList.forEach(usbDevice -> {
+            if (usbDevice.isUsbHub()) {
+                result.addAll(scanChildUsbDeviceInformation((UsbHub) usbDevice));
+            } else {
+                UsbDeviceDescriptor d = usbDevice.getUsbDeviceDescriptor();
+                short vendorId = d.idVendor();
+                short productId = d.idProduct();
+
+                String manufacturer = null;
+                String product = null;
+                String serialNumber = null;
+
+                /*
+                 * Note: the getString() calls below may fail depending on the Operating System:
+                 * - on Windows if no libusb device driver is installed for the device.
```

Comment:
I don't think this is correct. Without libusb you get nothing (even no vendorId or productId). But libusb is packaged as a native library with usb4java, so should be there on a number of platforms.

Reply:
Reading a bit further, there is a need for a driver for the device in Windows for libusb to be able to work with the device. And that may require installation if not installed already.

Reply:
It always finds vendorId and productId .. I tested it.